tests/rptest/tests/cluster_config_test.py

```diff
@@ -1855,6 +1855,9 @@ def __init__(self, *args, **kwargs):
 
         self.legacy_version = (23, 1)
 
+        # version where space_management legacy defaults are introduced
+        self.intermediate_version = (23, 2)
```

Comment:
Unrelated to this PR, for some reason I was under the impression Redpanda would complain/fail if we skipped any logical versions. I guess not though, given these other tests are already jumping from 23.1 to [23.3-soon-to-be-24.1](https://github.com/redpanda-data/redpanda/pull/16029)

Reply:
yeh good call out i think we should probably revisit the upgrade test infra if anything just to say that it still looks good